R/geom_venn.R

```diff
@@ -99,55 +99,55 @@ geom_venn <- function(mapping = NULL, data = NULL,
 }
 
 GeomVenn <- ggproto("GeomVenn", Geom,
-  required_aes = c("A", "B"),
-  optional_aes = c("C", "D", "label"),
-  extra_params = c("na.rm"),
-  setup_data = function(self, data, params) {
-    data %>% mutate(xmin = -2, xmax = 2, ymin = -2, ymax = 2)
-  },
-  draw_panel = function(self, data, panel_params, coord, ...) {
-    attr <- self$customize_attributes
-    sets <- c("A", "B", "C", "D")
-    sets <- sets[sets %in% names(data)]
-    show_elements <- NA
-    if ("label" %in% names(data)) {
-      show_elements <- "label"
-    }
-    venn <- prepare_venn_data(data, sets, show_elements)
-    d0 <- coord_munch(coord, venn$shapes, panel_params)
-    d <- d0 %>%
-      filter(!duplicated(group)) %>%
-      mutate(fill_color = attr$fill_color[group],
-             fill_alpha = attr$fill_alpha,
-             stroke_color = attr$stroke_color,
-             stroke_alpha = attr$stroke_alpha,
-             stroke_size = attr$stroke_size,
-             stroke_linetype = attr$stroke_linetype)
-    d1 <- coord_munch(coord, venn$labels, panel_params)
-    d2 <- coord_munch(coord, venn$texts, panel_params)
-    ggplot2:::ggname("geom_venn",
-      grobTree(
-        polygonGrob(
-          d0$x, d0$y, default.units = "native", id = d0$group,
-          gp = gpar(col = NA,
-                    fill = alpha(d$fill_color, d$fill_alpha))),
-        polygonGrob(
-          d0$x, d0$y, default.units = "native", id = d0$group,
-          gp = gpar(col = alpha(d$stroke_color, d$stroke_alpha),
-                    fill = NA,
-                    lwd = d$stroke_size * .pt,
-                    lty = d$stroke_linetype)),
-        textGrob(
-          self$set_names, d1$x, d1$y, default.units = "native",
-          hjust = d1$hjust, vjust = d1$vjust,
-          gp = gpar(col = attr$set_name_color,
-                    fontsize = attr$set_name_size * .pt)),
-        textGrob(
-          d2$text, d2$x, d2$y, default.units = "native",
-          hjust = d2$hjust, vjust = d2$vjust,
-          gp = gpar(col = attr$text_color,
-                    fontsize = attr$text_size * .pt))
-      )
-    )
-  }
+                    required_aes = c("A", "B"),
+                    optional_aes = c("C", "D", "label"),
+                    extra_params = c("na.rm"),
+                    setup_data = function(self, data, params) {
+                      data %>% mutate(xmin = -2, xmax = 2, ymin = -2, ymax = 2)
+                    },
+                    draw_panel = function(self, data, panel_params, coord, ...) {
+                      attr <- self$customize_attributes
+                      sets <- c("A", "B", "C", "D")
+                      sets <- sets[sets %in% names(data)]
+                      show_elements <- NA
```

Comment:
If `value_type` is added to `ggvenn()`, I strongly suggest adding the same parameter to `geom_venn()`.